redfish-core/include/utils/fan_utils.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "dbus_utility.hpp"
+#include "error_messages.hpp"
+
+#include <boost/system/error_code.hpp>
+
+#include <array>
+#include <memory>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <utility>
+
+namespace redfish
+{
+namespace fan_utils
+{
+constexpr std::array<std::string_view, 1> fanInterface = {
+    "xyz.openbmc_project.Inventory.Item.Fan"};
+constexpr std::array<std::string_view, 1> sensorInterface = {
+    "xyz.openbmc_project.Sensor.Value"};
+
+template <typename Callback>
+inline void getFanSensorsObject(const std::string& fanPaths,
+                                Callback&& callback)
+{
+    sdbusplus::message::object_path endpointPath{fanPath};
+    endpointPath /= "all_sensors";
+
+    dbus::utility::getAssociatedSubTree(
+        endpointPath,
+        sdbusplus::message::object_path("/xyz/openbmc_project/inventory"), 0,
+        sensorInterface,
+        [asyncResp, callback{std::forward<Callback>(callback)}](
```

Comment:
https://github.com/openbmc/bmcweb/commit/8cb2c024c4625e2fe2f0b107a865faffcd4bb770 suggests to use
```
[asyncResp, callback = std::forward<Callback>(callback)]
```